indico/web/client/js/react/components/ICSCalendarLink.jsx

```diff
@@ -9,46 +9,30 @@ import signURL from 'indico-url:core.sign_url';
 
 import PropTypes from 'prop-types';
 import React, {useReducer, useState} from 'react';
-import {Button, Dropdown, Icon, Input, Label, Grid, Popup, Header} from 'semantic-ui-react';
+import {Button, Icon, Input, Label, Grid, Popup, Header} from 'semantic-ui-react';
 
 import {Translate} from 'indico/react/i18n';
 import {indicoAxios, handleAxiosError} from 'indico/utils/axios';
 import {snakifyKeys} from 'indico/utils/case';
 
 import './ICSCalendarLink.module.scss';
 
-const initialState = {
-  open: false,
-  url: null,
-  text: null,
-  source: null,
-};
-
 function popupReducer(state, action) {
   switch (action.type) {
     case 'CLOSE':
       return {...state, url: null, open: false};
     case 'OPEN':
-      return {...state, open: true, text: action.text, source: action.source};
+      return {open: true, key: action.key, source: action.source};
     case 'LOADED':
       return {...state, url: action.url, source: null};
     default:
       return state;
   }
 }
 
-export default function ICSCalendarLink({
-  endpoint,
-  params,
-  renderButton,
-  dropdownPosition,
-  popupPosition,
-  options,
-  ...restProps
-}) {
-  const [popupState, dispatch] = useReducer(popupReducer, initialState);
+export default function ICSCalendarLink({endpoint, params, renderButton, popupPosition, options}) {
+  const [popupState, dispatch] = useReducer(popupReducer, {open: false});
```

Comment:
What's the reason for this change? Isn't it cleaner to keep any state fields we set in the reducer in the initial state as well?

Reply:
Sure, I can restore this if you feel like it. The reason is that it seems like boilerplate to me, all the fields were undefined (unused) except for `open` which is the only one we actually expect to use in the first state.

Reply:
at least i always saw "full" initial state definitions.. (and the advantage is that e.g. `something === null` would work, but with undefined keys that's not the case. also, there's always the full state visible in the react dev tools)